test/libsolidity/syntaxTests/metaTypes/name_other_contract.sol

```diff
@@ -2,6 +2,20 @@ contract Test {
     function f() public pure returns (string memory) {
         return type(C).name;
     }
+    function g() public pure returns (string memory) {
+        return type(A).name;
```

Comment:
Can you make this a semantic tests?

Reply:
I made semantic tests: https://github.com/ethereum/solidity/pull/9682/files#diff-69fbb0d4431b4d44ee5798751876dee6